src/app/public/modules/ag-grid/ag-grid-row-delete.directive.ts

```diff
@@ -0,0 +1,268 @@
+import {
+  AfterContentInit,
+  ChangeDetectorRef,
+  ComponentFactoryResolver,
+  ContentChild,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  ViewContainerRef
+} from '@angular/core';
+
+import {
+  SkyAffixAutoFitContext,
+  SkyAffixService,
+  SkyOverlayService
+} from '@skyux/core';
+
+import {
+  AgGridAngular
+} from 'ag-grid-angular';
+
+import {
+  RowNode
+} from 'ag-grid-community';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  SkyAgGridRowDeleteComponent
+} from './ag-grid-row-delete.component';
+
+import {
+  SkyAgGridRowDeleteCancelArgs
+} from './types/ag-grid-row-delete-cancel-args';
+
+import {
+  SkyAgGridRowDeleteConfig
+} from './types/ag-gird-row-delete-config';
+
+import {
+  SkyAgGridRowDeleteConfirmArgs
+} from './types/ag-grid-row-delete-confirm-args';
+
+import {
+  SkyAgGridRowDeleteContents
+} from './types/ag-grid-row-delete-contents';
+
+@Directive({
+  selector: '[skyAgGridRowDelete]'
+})
+export class SkyAgGridRowDeleteDirective implements AfterContentInit, OnDestroy {
+
+  /**
+   * Specifies the ids of the data in the rows on which inline delete's should be shown.
+   */
+  @Input()
+  set rowDeleteIds(value: string[]) {
+    this._rowDeleteIds = value;
+
+    if (!value) {
+      for (let config of this.rowDeleteConfigs) {
+        this.destroyRowDelete(config.id);
+      }
+
+      this.changeDetector.markForCheck();
+
+      return;
+    }
+
+    for (let id of value) {
+      const existingConfig = this.rowDeleteConfigs
+        .find(config => config.id === id);
+      if (existingConfig) {
+        existingConfig.pending = false;
```

Comment:
Nope. We show a wait on the inline delete when it is in its pending state. Because the inline delete does not fully go away when they confirm it - in order to allow the user to actually delete the row - we set it into pending. This logic ensures that if they pass back in a new array with the row still marked with an inline delete (as a sanity check) that we turn the pending back off. 

This is a little cleaner in `grid` as they have to message us to do this but I wanted to leave the logic here as otherwise there is no way to turn that pending state back off.